zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -639,20 +640,23 @@ private void createNote(NotebookSocket conn, HashSet<String> userAndRoles,
                           Notebook notebook, Message message)
       throws IOException {
     AuthenticationInfo subject = new AuthenticationInfo(message.principal);
-    Note note = notebook.createNote(subject);
-    note.addParagraph(); // it's an empty note. so add one paragraph
+    String noteName = null;
     if (message != null) {
-      String noteName = (String) message.get("name");
-      if (noteName == null || noteName.isEmpty()){
-        noteName = "Note " + note.getId();
-      }
-      note.setName(noteName);
+      noteName = (String) message.get("name");
     }
+    try {
+      Note note = notebook.createNote(subject, noteName);
+    
+      note.addParagraph(); // it's an empty note. so add one paragraph
 
-    note.persist(subject);
-    addConnectionToNote(note.getId(), (NotebookSocket) conn);
-    conn.send(serializeMessage(new Message(OP.NEW_NOTE).put("note", note)));
-    broadcastNoteList(subject);
+      note.persist(subject);
+      addConnectionToNote(note.getId(), (NotebookSocket) conn);
+      conn.send(serializeMessage(new Message(OP.NEW_NOTE).put("note", note)));
+      broadcastNoteList(subject);
+    } catch (DuplicateNameException dne) {
```

Comment:
For all websocket requests, we are sending responses from notebookServer class. So, i believe as a standard we should handle the response in the same file. So, rethrowing the exception wouldn't serve that purpose.